WordPress/Classes/Extensions/MediaService+Local.swift

```diff
@@ -0,0 +1,72 @@
+import Foundation
+import ImageIO
+
+/// Encapsulates Media functions relative to the local Media directory.
+///
+extension MediaService {
+
+    fileprivate static let mediaDirectoryName = "Media"
+
+    /// Returns filesystem URL for the local Media directory.
+    ///
+    class func localMediaDirectory() throws -> URL {
+        let fileManager = FileManager.default
+        let documents = fileManager.urls(for: .documentDirectory, in: .userDomainMask).first!
+        var media = documents.appendingPathComponent(mediaDirectoryName, isDirectory: true)
+        var isDirectory: ObjCBool = true
+        if fileManager.fileExists(atPath: media.path, isDirectory: &isDirectory) == false, isDirectory.boolValue == false {
```

Comment:
I've fixed this up actually, and added comments explaining what we're looking for.